libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1283,6 +1283,28 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         }
     }
 
+    /**
+     * Subscribes voice instructions trigger observer for all the necessary updates.
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun <T> registerVoiceInstructionsTriggerObserver(
+        observer: T
+    ) where T : RoutesObserver, T : RouteProgressObserver {
+        registerRoutesObserver(observer)
+        registerRouteProgressObserver(observer)
+    }
+
+    /**
+     * Unsubscribes voice instructions trigger observer from all the updates it was subsribed for.
+     */
+    @ExperimentalPreviewMapboxNavigationAPI
+    fun <T> unregisterVoiceInstructionsTriggerObserver(
+        observer: T
+    ) where T : RoutesObserver, T : RouteProgressObserver {
+        unregisterRoutesObserver(observer)
+        unregisterRouteProgressObserver(observer)
+    }
+
```

Comment:
With the current API user can:
1. Register anything what implements `RoutesObserver` and `RouteProgressObserver` as a `VoiceInstructionsTrigger`. It's getting challenging to change observers from which predownload logic takes data as users could start using this registration for a different purpose. 
2. Call `VoiceInstructionsDownloadTrigger#onRouteProgressChange` and `VoiceInstructionsDownloadTrigger#onRoutesChanged` that also limit our abilities in future changes as if we need more data for the new algorithm, we will have to support old usage despite the fact it's not right. 

What do you think about hiding those implementation details from the user so that users can't use the API incorrectly + we are able to change it in the future? What if you implement public API as `MapboxNavigationObserver` with a single callback `predownloadVoiceInstruction`? This way API surface will be less, i.e. users has less changes to use it wrong + we have more freedom modifying it in the future.
```suggestion

```

Reply:
Oh, basically the same as https://github.com/mapbox/mapbox-navigation-android/pull/6771/files#r1056539683
I am not resolving the conversation because it describes motivation for the same change from the different way. But let's discuss it in the Kyle's comment as he suggested it first.